bigquery_etl/docs/derived_datasets/generate_derived_dataset_docs.py

```diff
@@ -19,85 +19,108 @@
 SOURCE_URL = "https://github.com/mozilla/bigquery-etl/blob/generated-sql"
 
 
+def _get_metadata(table_path):
+    metadata = {}
+    metadata_file = table_path / METADATA_FILE
+    if metadata_file.exists():
+        with open(metadata_file) as stream:
+            try:
+                metadata = yaml.safe_load(stream)
+            except yaml.YAMLError as error:
+                print(error)
+    return metadata
+
+
+def _get_readme_content(table_path):
+    readme_file = table_path / README_FILE
+    if readme_file.exists():
+        return readme_file.read_text()
+
+
+def _get_referenced_tables_from_view(table_path):
+    referenced_tables = []
+    view_file = table_path / VIEW_FILE
+    if view_file.exists():
+        for referenced_table in extract_table_references(view_file.read_text()):
+            table_split = referenced_table.split(".")
+            if len(table_split) == 2:
+                # missing project ID, retrieve from file path
+                [dataset_id, table_id] = table_split
+                project_id = view_file.parent.parent.parent.name
+            elif len(table_split) == 3:
+                [project_id, dataset_id, table_id] = table_split
+            else:
+                continue
+
+            referenced_tables.append(
+                {
+                    "project_id": project_id,
+                    "dataset_id": dataset_id,
+                    "table_id": table_id,
+                }
+            )
+    return referenced_tables
+
+
+def _iter_table_markdown(table_paths, template):
+    for table_path in table_paths:
+        source_urls = {"Source Directory": f"{SOURCE_URL}/{str(table_path)}"}
+
+        referenced_tables = _get_referenced_tables_from_view(table_path)
+        if referenced_tables:
+            source_urls[
+                "View Definition"
+            ] = f"{SOURCE_URL}/{str(table_path / VIEW_FILE)}"
+
+        metadata = _get_metadata(table_path)
```

Comment:
Alternatively, the metadata could also be loaded via `Metadata.of_query_file(table_path)` or `Metadata.from_file(table_path / README_FILE)`
If the file doesn't exists, a `FileNotFound` exception will be raised.

Reply:
Gotcha, made a note to update this to use the `Metadata` API. We probably want to add some logging or show something when the metadata isn't present as well 👌